src/discord-api.ts

```diff
@@ -23,21 +23,29 @@ export const getAllTopics = async (env: Env, channelId?: string) => {
 	return channelId ? threads.filter((t) => t.parent_id === channelId) : threads
 }
 
-export const getMessages = async (
-	threads: { id: string; after?: string }[],
-	env: Env
-) => {
+const getMessages = async (threads: { id: string }[], env: Env) => {
 	const messages = (
 		await Promise.all(
 			threads.map(async (thread) => {
-				const messagesUrl = thread.after
-					? `${DISCORD_URL}/channels/${thread.id}/messages?after=${thread.after}&limit=100`
-					: `${DISCORD_URL}/channels/${thread.id}/messages?limit=100`
-
-				const messageResponse = await fetchData(messagesUrl, env.DISCORD_TOKEN)
-
-				const messages = (await messageResponse.json()) as APIMessage[]
-				return messages
+				let lastMessage = undefined
+				let fetchMore = true
+				do {
+					const messagesUrl = lastMessage
+						? `${DISCORD_URL}/channels/${thread.id}/messages?after=${lastMessage}&limit=100`
+						: `${DISCORD_URL}/channels/${thread.id}/messages?limit=100`
+
+					const messageResponse = await fetchData(
+						messagesUrl,
+						env.DISCORD_TOKEN
+					)
+
+					const messages = (await messageResponse.json()) as APIMessage[]
+					lastMessage = messages[messages.length - 1].id
```

Comment:
You could also use `at`
```suggestion
					lastMessage = messages.at(-1)?.id
```